javagen/src/main/java/com/azure/autorest/template/ConvenienceAsyncMethodTemplate.java

```diff
@@ -73,13 +111,8 @@ protected void writeInvocationAndConversion(
             String methodName = protocolMethod.getName();
             methodBlock.methodReturn(String.format("serviceClient.%1$s(%2$s)", methodName, invocationExpression));
         } else {
-            // return type is Mono<Void>, Response::getValue would be null
-            String returnExpression = responseBodyType.asNullable() == ClassType.Void
-                    ? "%1$s(%2$s).then()%3$s"   // return type is Mono<Void>, Response::getValue would be null
-                    : "%1$s(%2$s).map(Response::getValue)%3$s";
-
             methodBlock.methodReturn(
-                    String.format(returnExpression,
+                    String.format("%1$s(%2$s).flatMap(FluxUtil::toMono)%3$s",
```

Comment:
The only change is here. Rest is refactor.

Use `FluxUtil::toMono` so it will handle the case of `Response::getValue` is null.